src/services/dataStreamer.ts

```diff
@@ -128,54 +209,15 @@ export const streamData = async (
   opts: DataStreamOptions = {},
   req: Request
 ) => {
-  const timestamp = Date.now();
   const { reqId } = opts;
+  const outputStream = chain([stringer(), res]);
+  // Mutable mapping to track what assets are used for different pages
+  const assetData = {};
 
-  res.once('error', (err) => {
-    logger.error(createMessage(`Error with response pipeline: ${err}`, reqId));
-    // Destroy streams in hopes of preventing memory leaks
-    res.destroy();
-  });
+  streamTimestamp(outputStream, reqId);
+  await streamMetadata(outputStream, metadataCursor, reqId);
```

Comment:
wondering if we can do a Promise.all here but seems like streamAssets depends on the output of streamPages here from usage of assetData. perhaps can add a comment to explain this

Reply:
Added a comment!